extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java

```diff
@@ -131,16 +178,97 @@ public JWSAlgorithm getRecommendedAlgorithm(JWSSigner signer) {
+                var spec = new ECPublicKeySpec(pointQjce, priv.getParams());
+                pub = (ECPublicKey) KeyFactory.getInstance("EC").generatePublic(spec);
+
+            }
+
+            return new ECKey.Builder(Curve.forOID(curveName), pub).privateKey(priv).keyUse(KeyUse.SIGNATURE).build();
+
+        } catch (NoSuchAlgorithmException | InvalidParameterSpecException | InvalidKeySpecException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+    }
+
+
+    private OctetKeyPair convertEdDsaKey(KeyPair keypair) {
+        var pub = (EdECPublicKey) keypair.getPublic();
+        var priv = (EdECPrivateKey) keypair.getPrivate();
+
+        // if the public key is not present, an empty byte array is set, because as with all elliptic curves the public
+        // key can be recovered from the private key. OctetKeyPairs do this for us behind the scenes.
+        var urlX = ofNullable(pub).map(pubkey -> encodeX(pubkey.getPoint())).orElseGet(() -> Base64URL.encode(new byte[0]));
+        var urlD = ofNullable(priv).map(this::encodeD).orElse(null);
+
+        var curveName = ofNullable((EdECKey) priv).orElse(pub).getParams().getName();
+
+        return new OctetKeyPair.Builder(Curve.parse(curveName), urlX)
+                .d(urlD)
+                .build();
+    }
+
+    /**
+     * Encodes the private key part of an EdDSA key as {@link Base64URL}, throws an exception if the binary representation can't be obtained
+     */
+    @NotNull
+    private Base64URL encodeD(EdECPrivateKey edKey) {
+        var bytes = edKey.getBytes().orElseThrow(() -> new EdcException("Private key is not willing to disclose its bytes"));
+        return Base64URL.encode(bytes);
+    }
+
+
+    /**
+     * Encodes the public key part of an EdDSA key as {@link Base64URL}
+     */
+    @NotNull
+    private Base64URL encodeX(EdECPoint point) {
+        var bytes = reverseArray(point.getY().toByteArray());
+
+        // when the X-coordinate of the curve is odd, we flip the highest-order bit of the first (or last, since we reversed) byte
+        if (point.isXOdd()) {
+            var mask = (byte) 128; // is 1000 0000 binary
+            bytes[bytes.length - 1] ^= mask; // XOR means toggle the left-most bit
+        }
+
+        return Base64URL.encode(bytes);
+    }
+
+    /**
+     * reverses an array in-place
+     */
     private byte[] reverseArray(byte[] array) {
-        for (int i = 0; i < array.length / 2; i++) {
+        for (var i = 0; i < array.length / 2; i++) {
             var temp = array[i];
             array[i] = array[array.length - 1 - i];
             array[array.length - 1 - i] = temp;
         }
         return array;
     }
 
-    private Ed25519Verifier createOctetKeyPair(PublicKey publicKey) throws JOSEException {
+    private Ed25519Verifier createEdDsaSigner(PublicKey publicKey) throws JOSEException {
         var edKey = (EdECPublicKey) publicKey;
         var curveName = edKey.getParams().getName();
```

Comment:
not part of this PR, but I noticed that the process to get the `Curve` from the `EdECKey` (both public and private) is the same and it could be extracted in a common method

Reply:
easy fix. will do.